app/src/main/java/org/wikipedia/places/PlacesFragment.kt

```diff
@@ -464,27 +480,31 @@ class PlacesFragment : Fragment(), MapboxMap.OnMapClickListener {
 
         // Retrieve an unused bitmap from the pool
         val result = Glide.get(requireContext()).bitmapPool
-            .getDirty(MARKER_WIDTH, MARKER_HEIGHT, Bitmap.Config.ARGB_8888)
+            .getDirty(MARKER_SIZE, MARKER_SIZE, Bitmap.Config.ARGB_8888)
 
         // Make the background fully transparent.
         result.eraseColor(Color.TRANSPARENT)
 
         result.applyCanvas {
-            val srcRect = Rect(0, 0, thumbnailBitmap.width, thumbnailBitmap.height)
-
-            // Draw a filled circle that will serve as a mask for the thumbnail image.
-            drawCircle((MARKER_WIDTH / 2).toFloat(), (MARKER_WIDTH / 2).toFloat(),
-                ((MARKER_WIDTH / 2) - (MARKER_WIDTH / 16)).toFloat(), markerPaintSrc)
-
-            // Draw the thumbnail, which will be clipped by the circle mask above.
-            drawBitmap(thumbnailBitmap, srcRect, markerRect, markerPaintSrcIn)
-
-            // Draw the marker frame on top of the clipped thumbnail.
-            drawBitmap(markerBitmapBase, markerBitmapBaseRect, markerRect, null)
+            drawBitmap(circularBitmapWithBorder(thumbnailBitmap), null, markerRect, null)
         }
         return result
     }
 
+    private fun circularBitmapWithBorder(thumbnailBitmap: Bitmap? = null): Bitmap {
+        val radius = MARKER_SIZE / 2f
+        val outputBitmap = Bitmap.createBitmap(MARKER_SIZE, MARKER_SIZE, Bitmap.Config.ARGB_8888)
```

Comment:
This creates a `Bitmap` every time this function is called, and the `Bitmap` is never recycled. Bitmaps are special objects that must be [managed carefully](https://developer.android.com/topic/performance/graphics/manage-memory).

The only explicit bitmap instance should be `markerBitmapBase`.

Reply:
Not sure but can we use `Glide.get(requireContext()).bitmapPool.getDirty(MARKER_SIZE, MARKER_SIZE, Bitmap.Config.ARGB_8888)` just like what we did in the `getMarkerBitmap()`?